grammar

```diff
@@ -4,11 +4,14 @@
 program = declaration*
 declaration = function | variable_declr
 function = FUNC IDENTIFIER LEFT_BRACKET (type IDENTIFIER)* RIGHT_BRACKET type? block
+struct = STRUCT IDENTIFIER struct_block
 variable_declr = VAR IDENTIFIER type? EQUALS expression SEMICOLON
+field_declr = VAR IDENTIFIER type?
```

Comment:
Is `VAR` necessary here? The field declarations will only be used by structs so there would be no parsing benefit.

Reply:
I figured as fields are a kind of attribute it kinda makes sense - but they're by no means necessary. Up to you